### PR TITLE
Emphasize eval results metadata more

### PR DESCRIPTION
**Pull request by julien-c**
with screenshot from @lvwerra's [tweet](https://twitter.com/lvwerra/status/1439990571087671307)

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Very cool, thanks for the PR!

File: docs/hub/main.md [hunk 1]
```diff
@@ -15,15 +15,20 @@ Not one company, even the Tech Titans, will be able to “solve AI” by themsel
 
 The Hugging Face Hub hosts Git-based repositories which are storage spaces that can contain all your files 💾.
 
+The Hub currently hosts three different repo types:
+* models
+* datasets
+* spaces *which are ML demo apps*
```
Comment (stevhliu) [inline docs/hub/main.md:21]:
```suggestion
* Spaces *which are ML demo apps*
```

File: docs/hub/main.md [hunk 2]
```diff
@@ -32,7 +37,7 @@ On top of that, Hugging Face Hub repositories have many other advantages:
 
 ## What's a widget?
 
-Many repos have a widget that allows anyone to do inference directly on the browser!
+Many model repos have a widget that allows anyone to do inference directly on the browser!
```
Comment (stevhliu) [inline docs/hub/main.md:40]:
```suggestion
Many model repos have a widget that allows anyone to do inference directly in the browser!
```

File: docs/hub/model-repos.md [hunk 3]
```diff
@@ -50,8 +50,10 @@ Some useful information on them:
 * License identifiers are the keywords listed in the right column of [this table](#list-of-license-identifiers).
 * Dataset, metric, and language identifiers are those listed on the [Datasets](https://huggingface.co/datasets), [Metrics](https://huggingface.co/metrics) and [Languages](https://huggingface.co/languages) pages and in the [`datasets`](https://github.com/huggingface/datasets) repository.
 
+You can even specify your **model's eval results** in a structured way, which will allow the Hub to parse them, display them, and even link to Papers With Code leaderboards. Again the specification for this is [here](https://github.com/huggingface/huggingface_hub/blame/main/modelcard.md).
```
Comment (stevhliu) [inline docs/hub/model-repos.md:53]:
```suggestion
You can even specify your **model's eval results** in a structured way, which will allow the Hub to parse, display, and even link them to Papers With Code leaderboards. Specify this information [here](https://github.com/huggingface/huggingface_hub/blame/main/modelcard.md).
```

**Review by stevhliu — APPROVED** [review]
Looks great, thanks for highlighting this cool feature!

File: docs/hub/model-repos.md [hunk 4]
```diff
@@ -50,8 +50,10 @@ Some useful information on them:
 * License identifiers are the keywords listed in the right column of [this table](#list-of-license-identifiers).
 * Dataset, metric, and language identifiers are those listed on the [Datasets](https://huggingface.co/datasets), [Metrics](https://huggingface.co/metrics) and [Languages](https://huggingface.co/languages) pages and in the [`datasets`](https://github.com/huggingface/datasets) repository.
 
+You can even specify your **model's eval results** in a structured way, which will allow the Hub to parse, display, and even link them to Papers With Code leaderboards. Specify this information [here](https://github.com/huggingface/huggingface_hub/blame/main/modelcard.md).
```
Comment (julien-c) [inline docs/hub/model-repos.md:53]:
```suggestion
You can even specify your **model's eval results** in a structured way, which will allow the Hub to parse, display, and even link them to Papers With Code leaderboards. See how to format this data [in the metadata spec](https://github.com/huggingface/huggingface_hub/blame/main/modelcard.md).
```

Comment (julien-c) [reply docs/hub/model-repos.md:53]: changed the wording for this @stevhliu because what i mean is that the linked document *is* the spec

Comment (stevhliu) [reply docs/hub/model-repos.md]: Ah ok, my bad! For some reason I thought the link was to a model card `README.md` file on the Hub.

File: docs/hub/main.md [hunk 5]
```diff
@@ -15,15 +15,20 @@ Not one company, even the Tech Titans, will be able to “solve AI” by themsel
 
 The Hugging Face Hub hosts Git-based repositories which are storage spaces that can contain all your files 💾.
 
+The Hub currently hosts three different repo types:
+* models
+* datasets
+* Spaces *which are ML demo apps*
```
Comment (osanseviero) [inline docs/hub/main.md:21]:
```suggestion
* Spaces, *which are ML demo apps*
```

**Review by osanseviero — APPROVED** [review]
Thanks for this PR!

**Comment by lvwerra** [comment]
I was still working on my review when it was merged :) No objection to the proposed changes anyway, but some general thoughts on guiding users to make use of these features:
- What do you think about setting up a guide for nice model cards. Something along the lines of "How to make your models more impactful with good model cards". I think it could be fairly brief covering the most important points:
    - Describe how the model was trained
    - Create custom examples in the widget
    - Link to other resources such as GitHub repo/ Spaces etc.
    - Describe the data the model was trained on
    - ...
- We could add a message on the model card page when some information/metadata is missing. "It appears you have not added any [model metrics/languages/readme/...]. Checkout this guide (link to guide) to learn how to make your models stand out."
- Is something like a metadata GUI on the roadmap for the hub? Instead of having the user edit the Readme there could be a small GUI where one could add metadata fields.

What do you all think?

PS: that tweet got a lot of mileage :P

**Comment by osanseviero** [comment]
@lvwerra we have been adding metadata GUI in some sense. For example, users now pick the license during repo creation. Being able to easily modify metadata both programmatically and through the UX is part of the idea of the metadata validation [project](https://github.com/huggingface/moon-landing/projects/4#card-68902291) and now that we're wrapping up license, I think we'll go with language next.

As for invalid, users get YAML schema validation nowadays such as this when trying to save invalid schema.

> Error: YAML metadata schema issue on key "model-index" (key: /0/results/0/task must have property 'type' (error rule: properties))

See https://huggingface.co/ThomasSimonini/stable-baselines3-ppo-CartPole-v1 for example

If it's incomplete, users get something such as "Metadata error: specify a dataset to view leaderboard" https://huggingface.co/spacy/fr_dep_news_trf for example

As for guide, I indeed think it could improve. My only concern is duplication with https://huggingface.co/course/chapter4/4?fw=pt, but I agree with your idea :)

**Comment by lvwerra** [comment]
I didn't know about these existing efforts, thanks! Regarding the incomplete case: I was thinking about a softer message for optional metadata that's missing. More like a deletable banner saying "Your repo is missing a language tag. Learn how to add one here."

The course page is actually a good start. Maybe we can expand it a bit. Or is the course material supposed to be static until another version is released? cc @lewtun

**Comment by stevhliu** [comment]
I think repurposing the `Model Repos docs` to `How to make your models more impactful with good model cards` could also be an option, since that page focuses more heavily on the model card than the other features of the repo. This way we can be comprehensive and still cover all the options, instead of just several. 

I think combining the content from the course with the docs would also present a more complete guide for setting up a good model card. The course focuses more on describing the written sections (how a model was trained, data is was trained on, etc.) while the docs are more oriented towards all the metadata you can specify in the model card (widget examples/parameters, carbon footprint, etc).

**Pull request merged**